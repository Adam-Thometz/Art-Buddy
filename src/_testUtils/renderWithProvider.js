import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;