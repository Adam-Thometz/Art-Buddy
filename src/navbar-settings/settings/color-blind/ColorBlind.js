import { useSelector, useDispatch } from 'react-redux';
import { toggleColorBlind } from '_redux/settings/settingsActions';

import './ColorBlind.css';

import Toggle from '_components/toggle/Toggle';

const ColorBlind = () => {
  const { colorBlind } = useSelector(state => state.settings);
  const dispatch = useDispatch();

  const handleColorBlind = () => dispatch(toggleColorBlind());

  return (
    <div className='ColorBlind'>
      <Toggle currToggle={colorBlind} toggleFn={handleColorBlind} />
    </div>
  );
};

export default ColorBlind;