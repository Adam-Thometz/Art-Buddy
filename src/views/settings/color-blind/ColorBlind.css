.ColorBlind {
  display: flex;
  align-items: center;
}
