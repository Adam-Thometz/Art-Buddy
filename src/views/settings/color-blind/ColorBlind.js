import { useSelector, useDispatch } from "react-redux";
import { toggleColorBlind } from "store/settings/settingsReducer";

import "./ColorBlind.css";

import Toggle from "components/toggle/Toggle";

const ColorBlind = () => {
  const { colorBlind } = useSelector((state) => state.settings);
  const dispatch = useDispatch();

  const handleColorBlind = () => dispatch(toggleColorBlind());

  return (
    <div className="ColorBlind">
      <Toggle currToggle={colorBlind} toggleFn={handleColorBlind} />
    </div>
  );
};

export default ColorBlind;
