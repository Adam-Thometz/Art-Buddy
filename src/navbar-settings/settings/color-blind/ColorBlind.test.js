import React from "react";
import renderWithProvider from "_testUtils/renderWithProvider";
import ColorBlind from './ColorBlind';

describe('ColorBlind component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<ColorBlind />);
  });

  it('matches the snapshot of ColorBlind', () => {
    const { asFragment } = renderWithProvider(<ColorBlind />);
    expect(asFragment()).toMatchSnapshot();
  });
});