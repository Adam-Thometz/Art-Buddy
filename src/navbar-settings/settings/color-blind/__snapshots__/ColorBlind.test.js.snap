// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBlind component matches the snapshot of ColorBlind 1`] = `
<DocumentFragment>
  <div
    class="ColorBlind"
  >
    <div
      class="Toggle"
    >
      <div
        class="Toggle-switch off"
      >
        <div
          class="Toggle-button"
        />
        <span>
          ON
        </span>
        <span>
          OFF
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
